src/ppx_deriving_qcheck/ppx_deriving_qcheck.ml

```diff
@@ -1,243 +1,249 @@
 open Ppxlib
+module G = Qcheck_generators
+module O = G.Observable
 
-(** TypeGen can serve as a derivation environment. The map can be used
-    to remember how a type should be translated.
+(** {1. ppx_deriving_qcheck} *)
 
-    For instance, a recursive type must be derivated to a self recursive
-    call reducing the size.
+(** ppx_deriving_qcheck is a ppx deriver for QCheck generators. It does a traversal
+    map on type declarations annoted with [QCheck].
 
+    Example:
     {[
-    type tree = Leaf of int | Node of tree * tree
+    module Tree : sig
+      type t
+
+      val gen : t QCheck.Gen.t
```

Comment:
typo: `val gen_t`

Reply:
This is not a typo, but more a design choice. If type name is `t`, I do not add a suffix to the generator name. I don't know if it's a standard convention but I've seen multiple derivers do this.

Reply:
I suggest we document the behavior in the README.md.
Here's a suggestion to insert after line 19  https://github.com/c-cube/qcheck/blob/71b284f3fb7e5105d3ce2e84733126a77c16b13c/src/ppx_deriving_qcheck/README.md?plain=1#L19
(please fill-out/check/correct the derived names and types):
```
For `type tree` we derive two generators:
- `val gen_tree : tree Gen.t` and
- `val gen_tree_sized : ...`
For non-recursive types the latter is however not derived.

For types with the name `t` (`type t = ...`) which is a common idiom in OCaml code, 
the deriver omits the name from the derived generators, thus producing `val gen : ...` and
optionally `val gen_sized : ...`. 
```

Reply:
Added this in 29d41913c2edc0a52ef35fee667698c591b2debe, thanks for the suggestion :)

Reply:
Sorry for being unclear: can I ask you to fill in the dots `...`? 🙂

Reply:
Oh well yes, I thought you meant actual dots

Reply:
In principle the `...` can work.
However, I prefer being explicit about things that may seem obvious to us - because they may not be so for newcomers 🙏‍

Reply:
I kept "(`type t = ...`)"